database/src/postgres/models.rs

```diff
@@ -0,0 +1,569 @@
+use crate::schema::*;
+use diesel::prelude::*;
+use diesel_async::RunQueryDsl;
+
+/// State-indexer tables
+#[derive(Insertable, Queryable, Selectable)]
+#[diesel(table_name = state_changes_data)]
+pub struct StateChangesData {
+    pub account_id: String,
+    pub block_height: bigdecimal::BigDecimal,
+    pub block_hash: String,
+    pub data_key: String,
+    pub data_value: Option<Vec<u8>>,
+}
+
+impl StateChangesData {
+    pub async fn save(&self, mut conn: crate::postgres::PgAsyncConn) -> anyhow::Result<()> {
+        diesel::insert_into(state_changes_data::table)
+            .values(self)
+            .on_conflict_do_nothing()
+            .execute(&mut conn)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn get_state_key_value(
```

Comment:
I would expect to use it like this:

```rust
StateChangesData::get(...)
```

For me, `StateChangesData::get_state_key_value(...)` looks  
repetitive. Since, you're not just returning the model instance in this particular case I'd suggest something like `get_data_value`

For other models returning `Self`, I'd use `get()` as a name for the getter. What do you think?

Reply:
Diesel has already implemented a method called `get` which is used to get the values of a particular field. I also think that the current names reflect the essence of what is returned in queries.

Reply:
I definitely can live with that. I hope we do not end up in confusion later. I'll leave it to you. Decide what to do with this suggestion.